firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/internal/InstallActivity.java

```diff
@@ -53,23 +60,32 @@ protected void onResume() {
 
     if (!isUnknownSourcesEnabled()) {
       // See comment about install progress above. Same applies to unknown sources UI.
-      if (unknownSourceEnablementInProgress) {
+      if (alertDialog.isShowing()) {
         LogWrapper.getInstance()
             .e(
                 TAG
                     + "Unknown sources enablement was already in progress. It was either canceled or failed");
+        alertDialog.dismiss();
         finish();
         return;
       }
-
-      unknownSourceEnablementInProgress = true;
       showUnknownSourcesUi();
       return;
     }
 
     startAndroidPackageInstallerIntent();
   }
 
+  protected void onDestroy() {
+    super.onDestroy();
+    if (alertDialog.isShowing()) {
+      alertDialog.dismiss();
+      LogWrapper.getInstance()
+          .e(TAG + "Unknown sources enablement canceled. Activity was destroyed");
+      finish();
```

Comment:
Looks like there's no point in calling `finish()` from onDestroy. From the doc for onDestroy: "Perform any final cleanup before an activity is destroyed. This can happen either because the activity is finishing (someone called finish() on it), ..."

So onDestroy is a result of calling finish(). 

https://developer.android.com/reference/android/app/Activity#onDestroy()

Reply:
I was able to confirm this. Checking to see if window leaked error was on my side